hawkular-alerts-engine/src/main/resources/org/hawkular/alerts/engine/rules/ConditionMatch.drl

```diff
@@ -260,6 +264,33 @@ rule Rate
         insert( ce );
 end
 
+rule UpdateMissingStateFromData
+    when
+        $md : MissingState( $tenantId : tenantId, $tsource : source, $did : dataId, previousTime < time )
+        $d  : Data( tenantId == $tenantId, source == $tsource, id == $did)
+   then
+        if (log != null && log.isDebugEnabled()) {
+            log.debugf("UpdateMissingStateFromData: %s ", $md);
+        }
+        modify( $md ) {
+            setPreviousTime($d.getTimestamp()),
+            setTime($d.getTimestamp())
+        }

```

Comment:
I think this would do a couple more things:

1. `updatedMissingStates.add( $md )`
2. create a new, negative,` MissingConditionEval` and add it to WM